### PR TITLE
Macroatom restructure

### PR DESCRIPTION
**Pull request by Rodot-**
### :pencil: Restructures the MacroAtom to make it more independent from the plasma

**Type:** :rocket: `feature` 

The properties of the macro atom used for montecarlo/opacity purposes should be decoupled from the plasma, this PR attempts to restructure the macro atom workflow.

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

### CODE REVIEW

**Comment by andrewfullard** [comment]
I'm just seeing additions here and no deletions. Is the general idea to duplicate existing functionality?

**Comment by Rodot-** [comment]
> I'm just seeing additions here and no deletions. Is the general idea to duplicate existing functionality?

For the time being, yes, at least until I can guarantee everything works.

File: tardis/transport/montecarlo/base.py [hunk 1]
```diff
@@ -120,9 +114,8 @@ def initialize_transport_state(
 
         geometry_state = simulation_state.geometry.to_numba()
 
-        opacity_state = self.opacity_solver.solve(plasma)
         opacity_state_numba = opacity_state_to_numba(
-            opacity_state, self.opacity_solver.line_interaction_type
+            opacity_state, self.line_interaction_type
         )
         opacity_state_numba = opacity_state_numba[
             simulation_state.geometry.v_inner_boundary_index : simulation_state.geometry.v_outer_boundary_index
```
Comment (andrewfullard) [inline tardis/transport/montecarlo/base.py:121]: Suggests we perhaps need some slice handling by the solver? Or something similar. Not directly relevant to this PR.

File: tardis/opacities/macro_atom/macroatom_solver.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            Whether or not to initialize the transition probabilitiy coefficients and block references when solving the first time (default True)
+        normalize: bool
+            Whether or not to normalize the transition probabilities to unity. Default True
+        """
+
+        self.initialize = initialize
+        self.normalize = normalize
+
+    def initialize_non_markov_transition_probabilities(self, atomic_data):
+        """initialize the transition probabilitiy coefficients and block references when solving the first time
```
Comment (andrewfullard) [inline tardis/opacities/macro_atom/macroatom_solver.py:28]: Some minor typos in this docstring

File: tardis/opacities/macro_atom/base.py [hunk 3]
```diff
@@ -9,7 +9,231 @@
...
+
+    Returns
+    -------
+    np.ndarray
+        Reshaped macro atom transition probabilities
+    """
+    return macro_atom_data.transition_probability.values[np.newaxis].T
+
+
+def get_macro_atom_data(atomic_data):
```
Comment (andrewfullard) [inline tardis/opacities/macro_atom/base.py:212]: I think this shows us that we don't need *all* the atomic data to be passed around all the time.

**Review by andrewfullard — APPROVED** [review]

File: tardis/opacities/macro_atom/macroatom_solver.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        -------
+        tardis.opacities.macroatom_state.MacroAtomState
+            State of the macro atom ready to be placed into the OpacityState
+        """
+
+        transition_probabilities = (
+            self.solve_non_markov_transition_probabilities(
+                atomic_data,
+                legacy_plasma,
+                tau_sobolev,
```
Comment (wkerzendorf) [inline tardis/opacities/macro_atom/macroatom_solver.py:116]: that is part of the opacity state - I presume?

Comment (Rodot-) [reply tardis/opacities/macro_atom/macroatom_solver.py:116]: what do you mean?

File: tardis/opacities/macro_atom/base.py (hunk 3 above)
Comment (Rodot-) [reply tardis/opacities/macro_atom/base.py:212]: Definitely not, some kind of MacroAtomData structure could be useful

**Review by andrewfullard — DISMISSED** [review]

File: tardis/opacities/macro_atom/base.py [hunk 5]
```diff
@@ -137,5 +361,7 @@ def _get_macro_atom_data(atomic_data):
             return atomic_data.macro_atom_data_all
 
 
-class NonMarkovChainTransitionProbabilities(TransitionProbabilities):
+class NonMarkovChainTransitionProbabilities(
```
Comment (DeerWhale) [inline tardis/opacities/macro_atom/base.py:364]: This comment is referring to line 327 in the file (can't comment on the unmodified code) -- The original `initialize_macro_atom_transition_type_filters` function in the class has no returning value, is it intended?

Comment (DeerWhale) [reply tardis/opacities/macro_atom/base.py:364]: never mind, the properties is updated using self.xxxx .

File: tardis/opacities/opacity_solver.py [hunk 6]
```diff
@@ -2,9 +2,15 @@
...
 )
+from tardis.opacities.macro_atom.macroatom_solver import (
+    MacroAtomSolver,
+)
+from tardis.opacities.macro_atom.macroatom_state import MacroAtomState
 import numpy as np
 import pandas as pd
 
+from tardis.transport.montecarlo.configuration import montecarlo_globals
+
```
Comment (DeerWhale) [inline tardis/opacities/opacity_solver.py:13]: The three new imports is not used in this file, is this something preparing for the future refactor?

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**